lib/msal-browser/src/cache/BrowserCacheManager.ts

```diff
@@ -343,6 +343,32 @@ export class BrowserCacheManager extends CacheManager {
         this.removeAccountKeyFromMap(key);
     }
 
+    /**
+     * Extends inherited removeAccount function to include removal of cross-tenant account keys from the map
+     * @param account 
+     */
+    async removeAccountMultiTenant(key: string): Promise<void> {
+        const account: AccountEntity | null = this.getAccount(key);
+        
+        if (!account) {
+            this.logger.trace("BrowserCacheManager.removeAccountMultiTenant: called but no matching account was found in the cache.");
+            return;
+        }
+
+        const accountCacheKey = account.generateAccountKey();
+        super.removeAccount(accountCacheKey, true);
+
+        // Remove cross-tenant account keys from the map
+        const accountId = account.generateAccountId();
+        const accountKeys = this.getAccountKeys();
+        accountKeys.forEach((key) => {
+            if (key.indexOf(accountId) === 0) {
+                this.removeAccountKeyFromMap(key);
+            }
+        });
```

Comment:
Same question here, could this logic live in removeAccount?

Reply:
Moved it back into removeAccount using a new optional parameter.